runtime/ftplugin/query.lua

```diff
@@ -2,5 +2,29 @@
 -- Language:	Tree-sitter query
 -- Last Change:	2022 Mar 29
 
+if vim.b.did_ftplugin == 1 then
+  return
+end
+
+-- use treesitter over syntax
+vim.treesitter.start()
+
+-- query linter
+local buf = vim.api.nvim_get_current_buf()
+if not vim.b[buf].disable_query_linter then
+  local query_lint_on = vim.g.query_lint_on or { 'InsertLeave', 'CursorHoldI' }
```

Comment:
How is `b:disable_query_linter = true` different from `g:query_lint_on = {}`? Seems like they have the same effect, and we can remove `b:disable_query_linter`.

Reply:
The buffer variable is for disabling the linter on certain buffers only (like, say, the `inspect_tree`).

Reply:
If we change `query_lint_on` from a global variable to a buffer local variable, then I think that single variable could serve both cases.

Set defaults with a ftplugin file or FileType autocmd: `autocmd FileType query let b:lint_on = {'BufEnter'}`. Disable in a buffer with `let b:lint_on = {}`.

Reply:
ok, changing it back to what is was before...

Reply:
for me, `Filetype` seems to trigger after the filetype plugin.

Reply:
agreed on BufWrite+BufReadPre on Matrix chat